src/snowflake/snowpark/mock/plan.py

```diff
@@ -895,6 +909,215 @@ def calculate_expression(
         column = calculate_expression(exp.child, input_data, analyzer, expr_to_alias)
         column.sf_type = ColumnType(exp.to, exp.nullable)
         return column
+    if isinstance(exp, WindowExpression):
+        # Should return one column
+        window_function = exp.window_function
+        window_spec = exp.window_spec
+
+        # Process order by clause
+        if window_spec.order_spec:
+            res = input_data.sort_values(
+                by=[
+                    exp.child.name for exp in window_spec.order_spec
+                ],  # TODO: consider when the expr is something like col1+2
+                ascending=[
+                    isinstance(exp.direction, Ascending)
+                    for exp in window_spec.order_spec
+                ]
+                or True,
+            )
+        else:
+            res = input_data
+
+        res_index = res.index  # List of row indexes of the result
+
+        # Process partition_by clause
+        if window_spec.partition_spec:
+            res = res.groupby(
+                [exp.name for exp in window_spec.partition_spec],
+                sort=False,
+                as_index=False,
+            )
+            res_index = []
+            for r in res:
+                res_index += list(r[1].index)
```

Comment:
nit, can be simplified as

```python
res_index = [list(r[1].index) for r in res]
```

Reply:
This is for list concatenation. So I need `+=`.